storage/memory/memory.go

```diff
@@ -507,27 +579,29 @@ func (m *memory) TriplesForSubject(ctx context.Context, s *node.Node, lo *storag
 	if trpls == nil {
 		return fmt.Errorf("cannot provide an empty channel")
 	}
+
 	sUUID := UUIDToByteString(s.UUID())
 	m.rwmu.RLock()
 	defer m.rwmu.RUnlock()
 	defer close(trpls)
 
 	if lo.LatestAnchor {
-		lastTA := make(map[string]*time.Time)
-		trps := make(map[string]*triple.Triple)
-		for _, t := range m.idxS[sUUID] {
-			p := t.Predicate()
-			ppUUID := p.PartialUUID().String()
-			if p.Type() == predicate.Temporal {
-				ta, err := p.TimeAnchor()
-				if err != nil {
-					return err
-				}
-				if lta := lastTA[ppUUID]; lta == nil || ta.Sub(*lta) > 0 {
-					trps[ppUUID] = t
-					lastTA[ppUUID] = ta
-				}
-			}
+		if lo.FilterOptions != nil {
+			return fmt.Errorf("cannot have LatestAnchor and FilterOptions used at the same time inside lookup options")
+		}
+		lo.FilterOptions = &filter.StorageOptions{
+			Operation: filter.Latest,
+			Field:     filter.PredicateField,
+		}
+		// To guarantee that "lo.FilterOptions" will be cleaned at the driver level, since it was artificially created at the driver level for "LatestAnchor".
```

Comment:
optional: Given that this is done in multiple places it'd be useful to create a helper to clean up the filter options.
You can name it cleanFilterOptions and pass the "lo". Then code would read as:
defer cleanFilterOptions(lo)

Reply:
I like this idea and I would continue with it, but this is only being necessary here because this bool `LatestAnchor` in `lookupOptions` is not being deprecated yet. Then, for now, I decided to keep support on the OSS side of BadWolf for it too using a `FILTER latest predicate` behind the drapes, since they have the same behavior. But, since this bool will be deprecated soon, I tried to isolate it in just a block of code above `lo.FilterOptions != nil` so it will be easier to just delete it in the near future. Then, I am inclined to just let it as it is for now.